pkg/workloads/cortex/client/cortex/client.py

```diff
@@ -83,62 +84,69 @@ def deploy(
                 yaml.dump([api_spec], f)  # write a list
                 return self._deploy(cortex_yaml_path, force, wait)
 
-        project_dir = Path.home() / ".cortex" / "deployments" / str(uuid.uuid4())
-        with util.open_tempdir(str(project_dir)):
-            cortex_yaml_path = os.path.join(project_dir, "cortex.yaml")
-
-            if predictor is None:
-                # for deploying a traffic splitter
-                with open(cortex_yaml_path, "w") as f:
-                    yaml.dump([api_spec], f)  # write a list
-                    return self._deploy(cortex_yaml_path, force=force, wait=wait)
-
-            # Change if PYTHONVERSION changes
-            expected_version = "3.6"
-            actual_version = f"{sys.version_info.major}.{sys.version_info.minor}"
-            if actual_version < expected_version:
-                raise Exception("cortex is only supported for python versions >= 3.6")  # unexpected
-            if actual_version > expected_version:
-                is_python_set = any(
-                    conda_dep.startswith("python=") or "::python=" in conda_dep
-                    for conda_dep in conda_dependencies
-                )
-
-                if not is_python_set:
-                    conda_dependencies = [
-                        f"conda-forge::python={sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
-                    ] + conda_dependencies
-
-            if len(pip_dependencies) > 0:
-                with open(project_dir / "requirements.txt", "w") as requirements_file:
-                    requirements_file.write("\n".join(pip_dependencies))
-
-            if len(conda_dependencies) > 0:
-                with open(project_dir / "conda-packages.txt", "w") as conda_file:
-                    conda_file.write("\n".join(conda_dependencies))
-
-            if not inspect.isclass(predictor):
-                raise ValueError("predictor parameter must be a class definition")
-
-            with open(project_dir / "predictor.pickle", "wb") as pickle_file:
-                dill.dump(predictor, pickle_file)
-                if api_spec.get("predictor") is None:
-                    api_spec["predictor"] = {}
-
-                if predictor.__name__ == "PythonPredictor":
-                    predictor_type = "python"
-                if predictor.__name__ == "TensorFlowPredictor":
-                    predictor_type = "tensorflow"
-                if predictor.__name__ == "ONNXPredictor":
-                    predictor_type = "onnx"
-
-                api_spec["predictor"]["path"] = "predictor.pickle"
-                api_spec["predictor"]["type"] = predictor_type
+        if api_spec.get("name") is None:
+            raise ValueError("api_spec must be dictionary containing `name` key")
```

Comment:
nit: ```api_spec must have the `name` key set```

My only concern is that if `api_spec` is not passed in as a dictionary, the user could be confused. It might be worth checking for that separately, or if you want to avoid that, we could go with:

```api_spec must be a dictionary, and must have the `name` key set```